packages/generator/pyproject.toml

```diff
@@ -26,7 +26,7 @@ grpcio = "^1.49.1"
 protobuf = "^4.21"
 black = ">=24.8.0"
 click-option-group = ">=0.5.6"
-# ni-measurement-plugin-sdk-service = {version = "^2.0.0"}
+ni-measurement-plugin-sdk-service = {version = "^2.1.0-dev1", allow-prereleases = true}
```

Comment:
This PR is still absolutely necessary. Without this, the distribution package for `ni-measurement-plugin-sdk-generator` does not depend on the distribution package for `ni-measurement-plugin-sdk-service`. `poetry install` happens to work because of the dev dependencies, but real installation scenarios do not work.

Example:

```
PS D:\dev\measurement-plugin-python> cd .\packages\generator\
PS D:\dev\measurement-plugin-python\packages\generator> pipx install .
  installed package ni_measurement_plugin_sdk_generator 2.1.0.dev2, installed using Python 3.9.13
  These apps are now globally available
    - ni-measurement-plugin-client-generator.exe
    - ni-measurement-plugin-generator.exe
done! ✨ 🌟 ✨
PS D:\dev\measurement-plugin-python\packages\generator> cd ..\..\examples\client\
PS D:\dev\measurement-plugin-python\examples\client> dir
PS D:\dev\measurement-plugin-python\examples\client> ni-measurement-plugin-client-generator --help
Traceback (most recent call last):
  File "C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.13\lib\runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "C:\Users\Administrator\.pyenv\pyenv-win\versions\3.9.13\lib\runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "c:\users\administrator\.local\bin\ni-measurement-plugin-client-generator.exe\__main__.py", line 4, in <module>
  File "C:\Users\Administrator\pipx\venvs\ni-measurement-plugin-sdk-generator\lib\site-packages\ni_measurement_plugin_sdk_generator\client\__init__.py", line 11, in <module>
    from ni_measurement_plugin_sdk_service._internal.stubs.ni.measurementlink.measurement.v2 import (
ModuleNotFoundError: No module named 'ni_measurement_plugin_sdk_service'
```

Reply:
https://github.com/ni/measurement-plugin-python/pull/944

Reply:
Thanks for the PR, @bkeryan. Earlier, I didn't know we could have a package as a dependency under both "dependencies" and "dev.dependencies" and poetry would intelligently pick up the latest version.